control-plane/csi-driver/src/bin/controller/controller.rs

```diff
@@ -174,11 +175,41 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
         tracing::trace!(request = ?args);
         let _permit = self.create_volume_permit().await?;
 
-        if args.volume_content_source.is_some() {
-            return Err(Status::invalid_argument(
-                "Source for create volume is not supported",
-            ));
-        }
+        if let Some(source) = args.volume_content_source {
```

Comment:
And since its coming all the way from dataplane I don't think it would be `true` right?

Reply:
It may be during upgrade, if for some reason core gets updated before csi-controller. Then csi-controller will be completely broken and create a "normal" volume as we saw yesterday, which is really bad!

Reply:
Okay.